components/admin/BearerRefresh/BearerRefresh.tsx

```diff
@@ -0,0 +1,69 @@
+import { Button } from "@components/forms";
+import { logMessage } from "@lib/logger";
+import { BearerResponse, FormDBConfigProperties } from "@lib/types";
+import axios from "axios";
+import server from "i18next-hmr/server";
+import { useTranslation } from "next-i18next";
+import React, { useEffect, useState } from "react";
+
+export interface BearerRefreshProps {
+  form?: FormDBConfigProperties;
+}
+
+const BearerRefresh = (props: BearerRefreshProps): React.ReactElement => {
+  const { form } = props;
+  const [bearerTokenState, setBearerTokenState] = useState("");
+  const { t } = useTranslation("admin-templates");
+  const [errorState, setErrorState] = useState({ message: "" });
+
+  useEffect(() => {
+    setBearerTokenState(form?.bearerToken ? form.bearerToken : "");
+  }, []);
+
+  /**
+   * Connects to the database and refreshes the bearer token for the specified form ID.
+   *
+   * @param formID
+   */
+  const handleRefreshBearerToken = async (formID: number | undefined) => {
+    try {
+      setErrorState({ message: "" });
+      const serverResponse = await axios({
```

Comment:
When using a network call there should always be a Loading state so that the user knows that there is an action happening.  With an axios timeout set to `0` the component will wait indefinitely for a response and the user could continue to click the button repeatedly as no action is being demonstrated on the screen.

Reply:
Added a "Loader...".